File: AnelesAndroidProjects/app/src/main/java/za/co/anelesiguqa/anelesandroidprojects/MainActivity.java
package za.co.anelesiguqa.anelesandroidprojects;

import android.os.Bundle;
import android.widget.ListView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.recyclerview.widget.RecyclerView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.recycler_view_projects), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        RecyclerView list = findViewById(R.id.recycler_view_projects);
        Project[] projects = {
          new Project("Getting started App", "Our very first project", R.drawable.getting_started),
          new Project("Vin quote", "Motivational Quotes", R.drawable.quote),
          new Project("BMI calculator", "Calculates BMI", R.drawable.calculator),
          new Project("Inches converter", "Converts inches to meters", R.drawable.tape),
          new Project("The hungry developer", "Restaurant app", R.drawable.hungry_developer),
        };

        list.setAdapter(new ProjectsAdapter(projects));
    }
}

File: TheShamelessSelfPromoApp/app/src/main/java/za/co/anelesiguqa/selfpromoapp/MainActivity.kt
package za.co.anelesiguqa.selfpromoapp

import android.os.Bundle
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import za.co.anelesiguqa.selfpromoapp.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(binding.main) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        binding.buttonPreview.setOnClickListener {
            onPreviewClicked()
        }
    }

    private fun onPreviewClicked() {
        val contactName = binding.editTextContactName.text.toString()
        val contactNumber = binding.editTextContactNumber.text.toString()
        val myDisplayName = binding.editTextMyDisplayName.text.toString()
        val includeJunior = binding.checkBoxJunior.isChecked
        val jobTitle = binding.spinnerJobTitle.selectedItem?.toString()
        val immediateStart = binding.checkBoxImmediateStart.isChecked
        val startDate = binding.editTextStartDate.text.toString()

        val testString = "Contact Name: $contactName, Contact Number: $contactNumber, My Display Name: $myDisplayName," +
                " Include Junior: $includeJunior, Job Title: $jobTitle, Immediate Start: $immediateStart, Start Date: $startDate"
        Toast.makeText(this, testString, Toast.LENGTH_LONG).show()
    }

}